src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -115,7 +120,7 @@ public IReadOnlyList<LoggerClass> GetLogClasses(IEnumerable<ClassDeclarationSynt
                                             Name = methodSymbol.Name,
                                             Level = level,
                                             Message = message,
-                                            EventId = eventId,
+                                            EventId = (int)eventId.Value,
```

Comment:
Why do we need a cast to `(int)` here?

Reply:
A more performant way would be to use ``GetValueOrDefault`` instead of Value. It skips the second not null check, which was already done in .HasValue above